src/Cli/dotnet/NugetPackageDownloader/NuGetPackageDownloader.cs

```diff
@@ -218,7 +214,10 @@ public async Task<string> GetPackageUrl(PackageId packageId,
             SourceRepository repository = GetSourceRepository(source);
             if (repository.PackageSource.IsLocal)
             {
-                return Path.Combine(repository.PackageSource.Source, $"{packageId}.{resolvedPackageVersion}.nupkg");
+                return Path.Combine(
```

Comment:
This is probably feedback for a future PR, but using the FindPackagesByIdResource handles all of the logic in this method for you. 
It abstracts whether you're talking to a public or private feed, and it can help you download

Reply:
@edvilme, may you create an issue on the SDK to update some of our nuget api calls? Then we can decide how to prioritize that work in a different PR and you can resolve this comment.

Reply:
https://github.com/dotnet/sdk/issues/44169